### ISSUE TITLE
go-loops start to have errors, when a different function is defined with :clj-kondo/lint-as

### ISSUE THREAD

**Issue by stopachka** [clj-kondo/clj-kondo#2077]
**version**
v2023.04.14

**platform**

binary

**editor**

nvim

**problem**

I recently defined a macro and added `:clj-kondo/lint-as`. This works well for the macro, but I noticed that clj-kondo began to show false positives in go-loops. It says "function arguments should be wrapped in a vector", or "recur argument count mismatch". 

**repro**

Make sure to install core.async ([org.clojure/core.async "1.6.673"]) 

Try to run: 

```clojure
(ns testcljkondogoloop.core
  (:require [clojure.core.async :as a]))

(defn consume-call!
  "takes repeatedly from `ch` and calls `f`. stops when `ch` closes.

   work is done inside a go-try, so any error will return from the go block."
  [ch f]
  (a/go
    (loop []
      (when-some [v (a/<! ch)]
        (f v)
        (recur)))))

(defmacro consume!
  {:clj-kondo/lint-as 'clojure.core.let
   :doc "takes repeatedly from `ch`, binds the result to the symbol `v`
         and executes body. stops when `ch` closes, or if there is an error."}
  [[v ch] & body]
  `(consume-call! ~ch (fn [~v] ~@body)))

(a/go-loop []
  (a/<! (a/timeout 5000))
  (println "hello")
  (recur))

(defn foo
  "i don't do a whole lot."
  [x]
  (println x "hello, world!"))
```

```bash
λ clj-kondo --lint src/testcljkondogoloop/core.clj
src/testcljkondogoloop/core.clj::: error: More than one function overload with arity null.
src/testcljkondogoloop/core.clj:23:3: error: Function arguments should be wrapped in vector.
src/testcljkondogoloop/core.clj:24:3: error: Function arguments should be wrapped in vector.
src/testcljkondogoloop/core.clj:25:3: error: Function arguments should be wrapped in vector.
linting took 7ms, errors: 4, warnings: 0
```

**config**

I _think_ it's related to the `:clj-kondo/lint-as` metadata. When I remove it, the bug stops.

**expected behavior**

No errors for this go-loop

[ Optional: if you are interested in doing a PR yourself, please leave a note. ]
Sure!

**Comment by stopachka** [comment]
Ah! I wrote `'clojure.core.let`, where I should have written: `'clojure.core/let`
